package/network/services/lldpd/files/lldpd.init

```diff
@@ -42,80 +117,201 @@ write_lldpd_conf()
 
 	# Since lldpd's sysconfdir is /tmp, we'll symlink /etc/lldpd.d to /tmp/$LLDPD_CONFS_DIR
 	[ -e $LLDPD_CONFS_DIR ] || ln -s /etc/lldpd.d $LLDPD_CONFS_DIR
 }
 
-service_triggers() {
-	procd_add_reload_trigger "lldpd"
-}
 
 start_service() {
-
 	local enable_cdp
 	local enable_fdp
 	local enable_sonmp
 	local enable_edp
 	local lldp_class
-	local lldp_location
+	local lldpmed_no_inventory
+	local lldp_no_version
 	local readonly_mode
 	local agentxsocket
+	local filter
+
+	local enable_lldp
+	local force_lldp
+	local cdp_version
+	local force_cdp
+	local force_cdpv2
+	local force_edp
+	local force_fdp
+	local force_sonmp
 
 	config_load 'lldpd'
 	config_get_bool enable_cdp 'config' 'enable_cdp' 0
 	config_get_bool enable_fdp 'config' 'enable_fdp' 0
 	config_get_bool enable_sonmp 'config' 'enable_sonmp' 0
 	config_get_bool enable_edp 'config' 'enable_edp' 0
 	config_get lldp_class 'config' 'lldp_class'
-	config_get lldp_location 'config' 'lldp_location'
+	config_get_bool lldpmed_no_inventory 'config' 'lldpmed_no_inventory' 0
+	config_get_bool lldp_no_version 'config' 'lldp_no_version' 0
 	config_get_bool readonly_mode 'config' 'readonly_mode' 0
 	config_get agentxsocket 'config' 'agentxsocket'
+	config_get filter 'config' 'filter' 15
+
+	config_get_bool enable_lldp 'config' 'enable_lldp' 1
+	config_get_bool force_lldp 'config' 'force_lldp' 0
+	config_get cdp_version 'config' 'cdp_version' 'cdpv1v2'
+	config_get_bool force_cdp 'config' 'force_cdp' 0
+	config_get_bool force_cdpv2 'config' 'force_cdpv2' 0
+	config_get_bool force_edp 'config' 'force_edp' 0
+	config_get_bool force_fdp 'config' 'force_fdp' 0
+	config_get_bool force_sonmp 'config' 'force_sonmp' 0
 
-	mkdir -p /var/run/lldp
-	chown lldp:lldp /var/run/lldp
+	local readonly_mode
+	config_get_bool readonly_mode 'config' 'readonly_mode' 0
+
+	mkdir -p ${LLDPD_RUN}
+	chown lldp:lldp ${LLDPD_RUN}
 
 	# When lldpd starts, it also loads up what we write in this config file
 	write_lldpd_conf
 
 	procd_open_instance
-	procd_set_param command /usr/sbin/lldpd -d
+	procd_set_param command ${LLDPDBIN}
+	procd_append_param command -d # don't daemonize, procd will handle that for us
+
+	if [ $enable_lldp -gt 0 ]; then
+		if [ $force_lldp -gt 0 ]; then
+			procd_append_param command '-l'
+		fi
+	else
+		# Disable LLDP
+		procd_append_param command '-ll'
+	fi
+
+	if [ $enable_cdp -gt 0 ]; then
+		if [ $cdp_version == "cdpv1v2" ]; then
+			if [ $force_cdp -gt 0 ]; then
+				if [ $force_cdpv2 -gt 0 ]; then
+					# CDPv1 enabled, CDPv2 forced
+					procd_append_param command '-ccc'
+				else
+					# CDPv1 forced, CDPv2 enabled
+					procd_append_param command '-cc'
+				fi
+			else
+				# CDPv1 and CDPv2 enabled
+				procd_append_param command '-c'
+			fi
+		elif [ $cdp_version == "cdpv2" ]; then
+			if [ $force_cdp -gt 0 ]; then
+				# CDPv1 disabled, CDPv2 forced
+				procd_append_param command '-ccccc'
+			else
+				# CDPv1 disabled, CDPv2 enabled
+				procd_append_param command '-cccc'
+			fi
+		fi
+	fi
```

Comment:
Perhaps this could be simplified to:
```sh
if [ $enable_cdp -gt 0 ]; then
	if [ $cdp_version == "cdpv1v2" ]; then
		if [ $force_cdp -gt 0 ] && [ $force_cdpv2 -gt 0 ]; then
			procd_append_param command '-ccc'
		elif [ $force_cdp -gt 0 ]; then
			procd_append_param command '-cc'
		else
			procd_append_param command '-c'
		fi
	elif [ $cdp_version == "cdpv2" ]; then
		if [ $force_cdp -gt 0 ]; then
			procd_append_param command '-ccccc'
		else
			procd_append_param command '-cccc'
		fi
	fi
fi

```

Reply:
Maybe even something more aesthetic:
```sh
if [ $enable_cdp -gt 0 ]; then
	if [ $cdp_version == "cdpv2" ]; then
		if [ $force_cdp -gt 0 ]; then
			procd_append_param command '-ccccc'
		else
			procd_append_param command '-cccc'
		fi
	elif [ $cdp_version == "cdpv1v2" ]; then
		if [ $force_cdp -gt 0 ] && [ $force_cdpv2 -gt 0 ]; then
			procd_append_param command '-ccc'
		elif [ $force_cdp -gt 0 ]; then
			procd_append_param command '-cc'
		else
			procd_append_param command '-c'
		fi
	fi
fi

```